lib/core/tests/htmlrenderer.rs

```diff
@@ -1,30 +1,37 @@
 //! Test suite for the Web and headless browsers.
-
 #![cfg(target_arch = "wasm32")]
 
-extern crate wasm_bindgen_test;
-use wasm_bindgen_test::*;
+use web_test::web_configure;
+web_configure!(run_in_browser);
 
-wasm_bindgen_test_configure!(run_in_browser);
+use wasm_bindgen::prelude::wasm_bindgen;
+use wasm_bindgen::JsValue;
 
-pub mod common;
+#[wasm_bindgen(module = "/tests/bench_test.js")]
+extern "C" {
+    fn set_gradient_bg(
+        dom           : &JsValue,
+        red           : &JsValue,
+        green         : &JsValue,
+        blue          : &JsValue);
+}
 
 #[cfg(test)]
 mod tests {
-    use crate::common::TestContainer;
     use basegl::display::rendering::*;
     use basegl::system::web::StyleSetter;
-    use wasm_bindgen_test::*;
+    use basegl::system::web::get_performance;
+    use web_test::*;
+    use web_sys::Performance;
 
-    #[wasm_bindgen_test]
+    #[web_test(no_container)]
     fn invalid_container() {
         let scene = HTMLScene::new("nonexistent_id");
         assert!(scene.is_err(), "nonexistent_id should not exist");
     }
 
-    #[wasm_bindgen_test]
+    #[web_test]
     fn object_behind_camera() {
-        TestContainer::new("object_behind_camera", 320.0, 240.0);
         let mut scene = HTMLScene::new("object_behind_camera")
                                   .expect("Failed to create HTMLScene");
```

Comment:
As I've written above, let's jst always keep indentation in newline to be 4 spaces, not depending on previous line.